src/components/common/DatePicker.vue

```diff
@@ -1,7 +1,7 @@
 <template>
   <v-form :attach="attach" ref="form" class="date-picker-form">
     <v-menu v-model="displayPicker"
-            close-on-click
+            :close-on-click="false"
```

Comment:
This change prevents users from clicking away from the date picker to close it. (That behaviour left the date text field open.)
Now, the user must click Ok or Cancel. Either way, this component's parent can handle this and, say, hide the field.

(I verified with Cam that this was the original intent of having this prop, except the default value is True so you have to explicitly set it to False.)